mlflow/recipes/recipe.py

```diff
@@ -94,18 +94,20 @@ def run(self, step: str = None) -> None:
             last_executed_step_output_directory
         )
         if last_executed_step_state.status != StepStatus.SUCCEEDED:
+            last_step_error_mesg = (
```

Comment:
Q:
If `step` is None, does `last_executed_step_state` have valid value ?

Reply:
A good point. I had confused last_executed_step_state and last_executed_step. Let me add another check.

Reply:
The better thing here is to simply print the value of last_executed_step. If it is none, then None will be in the error message. Otherwise the string representation will be left to __str__ in the step implementation. This is better for custom steps that may want to surface additional details and does not rely on the contents of the BaseStep class.